### PR TITLE
Sync fork with upstream

### PR DESCRIPTION
**Pull request by npazosmendez**
There's only one conflict I still need to solve, but opening a draft PR so you can start reviewing.

### CODE REVIEW

**Comment by npazosmendez** [comment]
Also how do you handle de CLAssistant when merging commits from the upstream?

File: README.md [hunk 1]
```diff
@@ -111,7 +111,18 @@ Supported protocols
...
+| Golang Version | Last supported carbonapi version |
+|----------------|----------------------------------|
+| 1.10           | 0.10.0.1                         |
+| 1.12           | 0.12.6                           |
+| 1.16 / 1.17    | 0.15.6                           |
+| 1.18           | current                          |
+
+Overall rule of thumb is that carbonapi supports last 2 major go versions. E.x. at this moment Go 1.19 and 1.18 are supported.
+
+You can verify current versions that are being tested in [CI Configuration](https://github.com/go-graphite/carbonapi/blob/main/.github/workflows/tests.yml#L14).
```
Comment (carrieedwards) [inline README.md:125]: This link can be updated to our fork of the repo

File: expr/functions/legendValue/function.go [hunk 2]
```diff
@@ -31,14 +32,19 @@ func New(configFile string) []interfaces.FunctionMetadata {
 
 // legendValue(seriesList, newName)
 func (f *legendValue) Do(ctx context.Context, e parser.Expr, from, until int64, values map[parser.MetricRequest][]*types.MetricData) ([]*types.MetricData, error) {
-	arg, err := helper.GetSeriesArg(ctx, e.Args()[0], from, until, values)
+	arg, err := helper.GetSeriesArg(ctx, e.Arg(0), from, until, values)
 	if err != nil {
 		return nil, err
 	}
 
+<<<<<<< HEAD
```
Comment (carrieedwards) [inline expr/functions/legendValue/function.go:40]: Unresolved conflict here

**Comment by ywwg** [comment]
is it possible to squash your fixes to the merge into one commit since that's the most important part to review?

**Comment by ywwg** [comment]
(also willing to entertain reasons why you might not want to do that)

File: expr/functions/movingMedian/function.go [hunk 3]
```diff
@@ -128,27 +127,26 @@ func (f *movingMedian) Do(ctx context.Context, e parser.Expr, from, until int64,
...
-		r.Values = make([]float64, len(a.Values)-offset)
-		r.StartTime = (from + r.StepTime - 1) / r.StepTime * r.StepTime // align StartTime to closest >= StepTime
-		r.StopTime = r.StartTime + int64(len(r.Values))*r.StepTime
+		} else {
+			r.Values = make([]float64, len(a.Values)-offset)
+			r.StartTime = (from + r.StepTime - 1) / r.StepTime * r.StepTime // align StartTime to closest >= StepTime
+			r.StopTime = r.StartTime + int64(len(r.Values))*r.StepTime
 
-		data := movingmedian.NewMovingMedian(windowSize)
+			data := movingmedian.NewMovingMedian(windowSize)
```
Comment (carrieedwards) [inline expr/functions/movingMedian/function.go:135]: Looks like theres some weird spaces/tabbing here

**Comment by npazosmendez** [comment]
> is it possible to squash your fixes to the merge into one commit since that's the most important part to review?

That makes sense. I will see what I can do

File: expr/functions/legendValue/function.go (hunk 2 above)
Comment (npazosmendez) [reply expr/functions/legendValue/function.go:40]: Yep, I think this is the last one left. Anyway, I'll continue in the other PR https://github.com/grafana/carbonapi/pull/62

File: README.md (hunk 1 above)
Comment (npazosmendez) [reply README.md:125]: Nice catch, I'll address this in the other PR

**Pull request closed without merging**